File: java/src/deepspace/NumericDamage.java
/*
 * Here comes the text of your license
 * Each line should be prefixed with  * 
 */
package deepspace;
import java.util.ArrayList;
/**
 *
 * @author yabir
 */
public class NumericDamage extends Damage{
    private int nWeapons;
    
    NumericDamage(int w, int s){
        super(s);
        nWeapons = w;
    }
    
    NumericDamage(NumericDamage d){
        this(d.getNWeapons(), d.getNShields());
    }
    
    public int getNWeapons(){
	return nWeapons;
    }
    
    @Override
    public NumericDamage adjust(ArrayList<Weapon> wl,  ArrayList<ShieldBooster> sl){
        return new NumericDamage(Math.min(getNWeapons(), wl.size() ), Math.min(getNShields(), sl.size() ));
    }
    

    
    @Override
    public boolean hasNoEffect(){
        return super.hasNoEffect() && nWeapons == 0;
    }
    
    @Override
    NumericDamageToUI getUIversion(){
        return new NumericDamageToUI(this);
    }
    
    @Override
    public void discardWeapon(Weapon w){
        nWeapons -= 1;
    }
    
    @Override 
    NumericDamage copy(){
        return new NumericDamage(this);
    }
    
}

File: ruby/lib/NumericDamage.rb
# coding: utf-8
require_relative 'NumericDamageToUI'
require_relative 'WeaponType'
require_relative 'Damage'

module Deepspace
  class NumericDamage < Damage

    attr_reader :nWeapons #Número de armas a descartar
    
    def initialize(w, s)
      super(s)
      @nWeapons = w
    end

    def getUIversion
      NumericDamageToUI.new(self)
    end
    
    def self.newCopy
      new(nShields, nWeapons)
    end
        
    def adjust(wl, sl)
      NumericDamage.new([nWeapons,wl.length].min, adjustShields(sl))
    end
        
    def discardWeapon(w)
      @nWeapons -= 1
    end
    
    
    def hasNoEffect
      nShields == 0 && nWeapons == 0
    end

    def to_s
      "nWeapons: #{nWeapons};" + super
    end
    
    public :adjust
    public_class_method :new
  end
end
